lib/active_record/connection_adapters/oracle_enhanced/schema_dumper.rb

```diff
@@ -5,6 +5,12 @@ module ConnectionAdapters #:nodoc:
     module OracleEnhanced #:nodoc:
       class SchemaDumper < ConnectionAdapters::SchemaDumper #:nodoc:
         private
+          def column_spec_for_primary_key(column)
+            spec = super
+            spec.except!(:precision) if prepare_column_options(column) == { precision: "38", null: "false" }
```

Comment:
I'm not sure if the comparison with the hash object is the best way, but the dumped source has been fixed.
I have a one point. I think that the hash object instantiation can be reduced by making the hash object to be a constant.